guides/source/routing.md

```diff
@@ -529,23 +544,27 @@ If you wanted to link to just a magazine:
 <%= link_to 'Magazine details', @magazine %>
 ```
 
-For other actions, you just need to insert the action name as the first element of the array:
+For other actions, you need to insert the action name as the first element of the array, for `edit_magazine_ad_path`:
 
 ```erb
 <%= link_to 'Edit Ad', [:edit, @magazine, @ad] %>
 ```
 
 This allows you to treat instances of your models as URLs, and is a key advantage to using the resourceful style.
 
+NOTE: In order to automatically derive paths and URLs from objects such as `[@magazine, @ad]`, Rails uses methods from `ActiveModel::Naming` and `ActiveModel::Conversion` modules. Specifically the `@magazine.model_name.route_key` returns `magazines` and `@magazine.to_param` returns a string representation of the model's `id`. So the generated path may be something like `/magazines/1/ads/42` for the objects `[@magazine, @ad]`.
```

Comment:
Let's add links for the `ActiveModel::Naming` and `ActiveModel::Conversion` modules